nvflare/recipe/fedavg.py

```diff
@@ -0,0 +1,247 @@
+
+        # Setup aggregator
+        if self.aggregator is None:
+            self.aggregator = InTimeAccumulateWeightedAggregator(expected_data_kind=self.aggregator_data_kind)
+        else:
+            if not isinstance(self.aggregator, Aggregator):
+                raise ValueError(f"Invalid aggregator type: {type(self.aggregator)}. Expected type: {Aggregator}")
+
+        # Add shareable generator and aggregator
+        shareable_generator = FullModelShareableGenerator()
+        shareable_generator_id = job.to_server(shareable_generator, id="shareable_generator")
+        aggregator_id = job.to_server(self.aggregator, id="aggregator")
+
+        # Add controller
+        controller = ScatterAndGather(
+            min_clients=self.min_clients,
+            num_rounds=self.num_rounds,
+            wait_time_after_min_received=0,
+            aggregator_id=aggregator_id,
+            persistor_id=persistor_id,
+            shareable_generator_id=shareable_generator_id,
+        )
+        job.to_server(controller)
+
+        # Add client executors
+        if isinstance(self.train_args, dict):
+            # Per-client configuration
+            for site_name, site_args in self.train_args.items():
+                executor = ScriptRunner(
+                    script=self.train_script,
+                    script_args=site_args,
+                    launch_external_process=self.launch_external_process,
+                    command=self.command,
+                    framework=self.framework,
+                    server_expected_format=self.server_expected_format,
+                    params_transfer_type=self.params_transfer_type,
+                )
+                job.to(executor, site_name)
+        else:
+            # Unified configuration
+            executor = ScriptRunner(
+                script=self.train_script,
+                script_args=self.train_args,
+                launch_external_process=self.launch_external_process,
+                command=self.command,
+                framework=self.framework,
+                server_expected_format=self.server_expected_format,
+                params_transfer_type=self.params_transfer_type,
+            )
+            job.to_clients(executor)
+
+        Recipe.__init__(self, job)
+
+    def _get_analytics_receiver(self):
+        """Get the analytics receiver for this recipe.
+
+        Returns:
+            AnalyticsReceiver or None: The analytics receiver to use, or None for no analytics.
+
+        Note:
+            Child classes (PT/TF wrappers) can override this to provide framework-specific defaults.
+        """
+        return None
+
+    def _setup_model_and_persistor(self, job: BaseFedJob) -> str:
+        """Setup framework-specific model components and persistor.
+
+        Returns:
+            str: The persistor_id to be used by the controller.
+        """
+        return job.to_server(self.model_persistor, id="persistor")
```

Comment:
you might want to add a check here to make sure model_persistor is not None then we add it.